pkg/yaml/yaml.go

```diff
@@ -67,6 +67,8 @@ func Marshal(v interface{}) ([]byte, error) {
 		yamlEncoder := yaml.NewEncoder(
 			&b,
 			yaml.Indent(2),
+			yaml.UseSingleQuote(true),
+			yaml.UseLiteralStyleIfMultiline(true),
```

Comment:
@yxxhero To be extra sure- would u mind clarifying why do we need this?

Reply:
https://github.com/helmfile/helmfile/blob/219602ebc776274dcb5f1fef98d2f780b9e863c2/test/e2e/template/helmfile/testdata/snapshot/oci_need/input.yaml#L16-L23

because of the case above. if no `yaml.UseLiteralStyleIfMultiline(true)`, Multiline render error.

Reply:
I see! Thanks a lot for your clarification.